backend/persistent/datastores/csv.js

```diff
@@ -0,0 +1,92 @@
+const alasql = require('alasql');
+import fetch from 'node-fetch';
+import papa from 'papaparse';
+import {type} from 'ramda';
+
+import {parseSQL} from '../../parse';
+
+// connect() stores the data parsed from CSV files here
+const connectionData = {};
+
+// define type error thrown by this connector
+export function CSVError(url, errors) {
```

Comment:
You created your own CSV Error with specific information.  I am just trying to clarify why you needed your own error type.  How will this be handled on the client?